upup/pkg/fi/vfs_castore.go

```diff
@@ -634,20 +634,22 @@ func formatFingerprint(data []byte) string {
 }
 
 func insertFingerprintColons(id string) string {
-	var buf bytes.Buffer
+	remaining := id
 
+	var buf bytes.Buffer
 	for {
-		if id == "" {
+		if remaining == "" {
 			break
 		}
 		if buf.Len() != 0 {
 			buf.WriteString(":")
 		}
-		if len(id) < 2 {
-			buf.WriteString(id)
+		if len(remaining) < 2 {
+			glog.Warningf("unexpected format for SSH public key id: %q", id)
```

Comment:
Should we just err here? Do we have a use case where this should work?

Reply:
This mostly catches instances where I was double-encoding, and warns the developer.  The fingerprint comes from code, and I didn't want to change the method signature just for a nice-to-have.

The other change is to rename id to remaining.